--defsym=__rtc_localtime=$(shell powershell [int](Get-Date -UFormat +%s)[0]) not accepted

When using the prebuilt visualTeensy 1.0.0.0 example (Win10, 1909), I get the following error message:

`[LD]  .vsteensy/build/visualTeensyTest.elf 
c:/progra~2/arduino/hardware/tools/arm/bin/../lib/gcc/arm-none-eabi/5.4.1/../../../../arm-none-eabi/bin/ld.exe:--defsym:1: syntax error
collect2.exe: error: ld returned 1 exit status
make: *** [makefile:236: .vsteensy/build/visualTeensyTest.elf] Error 1
The terminal process "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command & 'C:/Program Files (x86)/VisualTeensy/make.exe' all -j -Otarget" terminated with exit code: 1.`

It looks like ld.exe doesn't like the syntax of the --defsym flag. When replacing the whole $(...) part with 1 (result given by the powershell), it doesn't complain at all.

Is that a Windows or a visualTeensy issue? Thanks!

--- a/vtCore/Implementation/BoardsTxtParser/Board.cs
+++ b/vtCore/Implementation/BoardsTxtParser/Board.cs
@@ -56,25 +56,19 @@ namespace vtCore
 
             if (ldFlagOpt.Key != null)
             {
-                string oldVal = ldFlagOpt.Value.Replace("\"", "");
-                string newVal;
-
-                if(oldVal.Contains("-Wl,"))
+                string ldFlags = ldFlagOpt.Value.Replace("\"", "").Replace("{build.core.path}", "$(CORE_BASE)");
+              
+                if (ldFlags.Contains("-Wl,"))
                 {
-                    oldVal = oldVal.Replace("-Wl,","-Wl,--print-memory-usage,");
+                    ldFlags = ldFlags.Replace("-Wl,", "-Wl,--print-memory-usage,");
+                    if (ldFlags.Contains("--defsym=__rtc_localtime"))
+                    {
+                        ldFlags = ldFlags.Replace("{extra.time.local}", "$(TIME_SYM)");
+                    }
                 }
-                if (oldVal.Contains("--defsym=__rtc_localtime"))
-                {
-                    newVal = oldVal.Replace("{extra.time.local}", "$(shell powershell [int](Get-Date -UFormat +%s)[0])");
-                }
-                else
-                {
-                    int pos = oldVal.IndexOf("--relax") + 7;
-                    newVal = oldVal.Insert(pos, ",--defsym=__rtc_localtime=$(shell powershell [int](Get-Date -UFormat +%s)[0])");
-                }
-                
+              
                 allOptions.Remove("build.flags.ld");
-                allOptions.Add("build.flags.ld", newVal);
+                allOptions.Add("build.flags.ld", ldFlags);
             }
 
             return allOptions;

--- a/vtCore/Implementation/Project Generators/Makefiles/Makefile_Make.cs
+++ b/vtCore/Implementation/Project Generators/Makefiles/Makefile_Make.cs
@@ -77,10 +77,17 @@ namespace vtCore
             if (project.debugSupport != DebugSupport.none && !String.IsNullOrWhiteSpace(setup.uplCLIBase.path)) mf.Append($"UPL_CLICMD_B     := {setup.uplCLIBase.shortPath}\n");
 
 
+
             mf.Append("\n#******************************************************************************\n");
             mf.Append("# Flags and Defines\n");
             mf.Append("#******************************************************************************\n");
 
+            if (makeEntry("dummy", "build.flags.ld", options).Contains("TIME_SYM"))
+            {
+                mf.Append("TIME_SYM    := $(shell powershell [int][double]::Parse((Get-Date -UFormat %s)))" + "\n");
+            }
+
+            mf.Append("\n");
             mf.Append(makeEntry("FLAGS_CPU   := ", "build.flags.cpu", options) + "\n");
             mf.Append(makeEntry("FLAGS_OPT   := ", "build.flags.optimize", options) + "\n");
             mf.Append(makeEntry("FLAGS_COM   := ", "build.flags.common", options) + makeEntry(" ", "build.flags.dep", options) + "\n");
@@ -90,13 +97,11 @@ namespace vtCore
             mf.Append(makeEntry("FLAGS_CPP   := ", "build.flags.cpp", options) + "\n");
             mf.Append(makeEntry("FLAGS_C     := ", "build.flags.c", options) + "\n");
             mf.Append(makeEntry("FLAGS_S     := ", "build.flags.S", options) + "\n");
-
-            mf.Append(makeEntry("FLAGS_LD    := ", "build.flags.ld", options).Replace("{build.core.path}", "$(CORE_BASE)") + "\n");
+            mf.Append(makeEntry("FLAGS_LD    := ", "build.flags.ld", options) + "\n");
 
             mf.Append("\n");
             mf.Append(makeEntry("LIBS        := ", "build.flags.libs", options) + "\n");
-            //mf.Append(makeEntry("LD_SCRIPT   := ", "build.mcu", options) + ".ld\n");
-
+            
             mf.Append("\n");
             mf.Append(makeEntry("DEFINES     := ", "build.flags.defs", options) + makeEntry(" -DARDUINO_", "build.board", options) + " -DARDUINO=10813\n");
             mf.Append("DEFINES     += ");
@@ -151,7 +156,7 @@ namespace vtCore
             //}
             //else
             //{
-                mf.Append("USR_SRC         := src\n");
+            mf.Append("USR_SRC         := src\n");
             //}
             mf.Append("LIB_SRC         := lib\n");
             mf.Append("CORE_SRC        := $(CORE_BASE)\n\n");
@@ -195,12 +200,12 @@ namespace vtCore
 
             mf.Append("\n");
             mf.Append("#User Sources -----------------------------------------------------------------\n");
-  //          if (cfg.setupType == SetupTypes.expert && project.buildSystem == BuildSystem.makefile && project.useInoFiles) mf.Append("USR_INO_FILE    := $(USR_SRC)/$(TARGET_NAME).ino\n");
+            //          if (cfg.setupType == SetupTypes.expert && project.buildSystem == BuildSystem.makefile && project.useInoFiles) mf.Append("USR_INO_FILE    := $(USR_SRC)/$(TARGET_NAME).ino\n");
             mf.Append("USR_C_FILES     := $(call rwildcard,$(USR_SRC)/,*.c)\n");
             mf.Append("USR_CPP_FILES   := $(call rwildcard,$(USR_SRC)/,*.cpp)\n");
             mf.Append("USR_S_FILES     := $(call rwildcard,$(USR_SRC)/,*.S)\n");
             mf.Append("USR_OBJ         := $(USR_S_FILES:$(USR_SRC)/%.S=$(USR_BIN)/%.o) $(USR_C_FILES:$(USR_SRC)/%.c=$(USR_BIN)/%.o) $(USR_CPP_FILES:$(USR_SRC)/%.cpp=$(USR_BIN)/%.o)\n");
-  //          if (cfg.setupType == SetupTypes.expert && project.buildSystem == BuildSystem.makefile && project.useInoFiles) mf.Append("USR_OBJ         += $(USR_INO_FILE:$(USR_SRC)/%.ino=$(USR_BIN)/%.o)\n");
+            //          if (cfg.setupType == SetupTypes.expert && project.buildSystem == BuildSystem.makefile && project.useInoFiles) mf.Append("USR_OBJ         += $(USR_INO_FILE:$(USR_SRC)/%.ino=$(USR_BIN)/%.o)\n");
 
             mf.Append("\n");
             mf.Append("# Core library sources --------------------------------------------------------\n");
@@ -334,7 +339,7 @@ namespace vtCore
             mf.Append("\n");
             mf.Append("$(LIB_BIN)/%.o: $(LIBS_LOCAL_BASE)/%.c\n");
             mf.Append("\t@echo $(COL_LIB)LIB [CC]  $(notdir $<) $(COL_ERR)\n");
-            mf.Append("\t@\"$(CC)\" $(C_FLAGS) $(INCLUDE) -o $@ -c $<\n") ;
+            mf.Append("\t@\"$(CC)\" $(C_FLAGS) $(INCLUDE) -o $@ -c $<\n");
 
             mf.Append("\n");
             mf.Append("# Handle user sources ---------------------------------------------------------\n");

